xdsl/dialects/tensor.py

```diff
@@ -216,7 +218,14 @@ def parse(cls, parser: Parser) -> Self:
 
 
 ReassociationAttr = Annotated[
-    ArrayAttr[ArrayAttr[IntegerAttr[I64]]],
+    ArrayAttr[
+        ArrayAttr[
+            Annotated[
+                IntegerAttr[I64],
+                IntegerAttr.constr(value=IntAttrConstraint(AtLeast(0))),
```

Comment:
Could you rebase onto the latest main? I worry that this will trigger the `AnyOf` disjointness check

Reply:
I was worried over nothing, this corresponds to an `AllOf` rather than an `AnyOf`